extension/src/test/Events.test.ts

```diff
@@ -0,0 +1,33 @@
+import * as assert from "assert";
+import * as path from "path";
+import * as vscode from "vscode";
+import { xliffCache } from "../Xliff/XLIFFCache";
+
+suite("Events Tests", () => {
+  const cachedFilePath = path.join(
+    __dirname,
+    "../../",
+    "src/test/resources/NAB_AL_Tools.da-DK.xlf"
+  );
+  test("NABFunctions.onDidChangeTextDocument", async function () {
+    assert.ok(xliffCache.size === 0, "Cache is not empty.");
+    const textDocument = await vscode.workspace.openTextDocument(
+      cachedFilePath
+    );
+    const editor = await vscode.window.showTextDocument(textDocument);
+    editor.edit((editBuilder) => {
+      editBuilder.insert(new vscode.Position(0, 0), "    ");
```

Comment:
Could it be worth the effort to edit data in a target element, and check the XliffDocument from cache that the value is updated correctly in cache?

Reply:
Yes, that's probably the single biggest flaw the cache could have so we should add that.

Reply:
new issue for this, or add this in this PR?

Reply:
I'll add it in this later today

Reply:
done